site/src/scaffold/scripts/popover.ts

```diff
@@ -0,0 +1,72 @@
+export function initPopovers(container: HTMLElement) {
+	container.addEventListener('click', (event: Event) => {
+		let targetPopover: HTMLDetailsElement | null;
+		targetPopover =
+			event.target instanceof Element &&
+			(event.target.closest('details.popover') as HTMLDetailsElement)
+				? (event.target.closest('details.popover') as HTMLDetailsElement)
+				: null;
+
+		if (!targetPopover) {
+			targetPopover =
+				event.target instanceof Element &&
+				event.target.shadowRoot &&
+				event.target.shadowRoot.activeElement &&
+				(event.target.shadowRoot.activeElement.closest('details.popover') as HTMLDetailsElement)
+					? (event.target.shadowRoot.activeElement.closest('details.popover') as HTMLDetailsElement)
+					: null;
+		}
+
+		if (!targetPopover) {
+			return;
+		}
+
+		const allPopovers: HTMLDetailsElement[] = Array.from(
+			container.querySelectorAll('details.popover')
+		);
+
+		if (!targetPopover.open) {
+			const keyHandler = (event: KeyboardEvent) => {
+				if (event.key === 'Escape') {
+					closePopovers();
+				}
+			};
+
+			const clickHandler = (event: Event) => {
+				if (!(event.target instanceof Element)) {
+					return;
+				}
+
+				if (!targetPopover?.contains(event.target)) {
+					targetPopover ? closePopovers([targetPopover]) : closePopovers();
```

Comment:
I'm not certain about this, and it seems fine to me, but I have a feeling the `no unused expresssion` lint rules that are enabled in other repos will complain about it. solution would just be to un-nest and use `if else` (I think). Might be worth pasting this into another repo and running lint.

Reply:
Refactored that section, don't think it's relevant anymore.